docs/error_codes.yaml

```diff
@@ -0,0 +1,151 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# This file contains structured documentation about the Pelican parameters.
+# While it is somewhat human-readable, it is meant to help with the documentation
+# generation.
+
+############################
+#    Client Error Codes    #
+############################
+
+############################
+#     Parameter Errors     #
+############################
+---
+type: Parameter
+code: 1000
+clientExitCode: 4
+description: >-
+  If the client failed to start, or was started with invalid parameters, or otherwise believes what it
+  was asked to do is impossible or the request itself is malformed.
+retryable: false
+
+############################
+#    Resolution Errors     #
+############################
+---
+type: Resolution
+code: 2000
+clientExitCode: 5
+description: >-
+  Indicates that the client failed to even attempt to contact the server
+retryable: false
+
+############################
+#      Contact Errors      #
+############################
+---
+type: Contact
+code: 3000
+clientExitCode: 6
+description: >-
+  The client attempted to contact the server at its resolved address and failed to do so
+retryable: false
+---
+type: Contact.Director
+code: 3001
+clientExitCode: 6
+description: >-
+  The client attempted to contact the director at its found address but failed to do so
+retryable: false
+---
+type: Contact.Cache
+code: 3002
+clientExitCode: 11
+description: >-
+  The client attempted to contact the cache but failed to do so
+retryable: true
+---
+type: Contact.Origin
+code: 3003
+clientExitCode: 6
+description: >-
+  The client attempted to contact the origin but failed to do so
+retryable: false
+---
+type: Contact.Registry
+code: 3004
+clientExitCode: 6
+description: >-
+  The client attempted to contact the registry (usually through the director) but failed to do so
+retryable: false
+
+############################
+#   Authorization Errors   #
+############################
+---
+type: Authorization
+code: 4000
+clientExitCode: 7
+description: >-
+  The client contacted the server but failed to authenticate, or failed to authorize, or if
+  the server replied with an authorization error when the file was requested or sent.
+  Basically a HTTP/403 error
+retryable: false
+
+############################
+#   Specification Errors   #
+############################
+---
+type: Specification
+code: 5000
+clientExitCode: 8
+description: >-
+  If the client successfully contacted the server and received a definitive response that the
+  desired file was not present or could not be created. Usually the submitters fault
+retryable: false
+---
+type: Specification.FileNotFound
+code: 5001
+clientExitCode: 8
+description: >-
+  If the client successfully contacted the server but the desired file does not exist for download.
```

Comment:
Can you modify the other descriptions to also include a period if they don't have one?